src/Gjallarhorn/State.fs

```diff
@@ -82,3 +83,34 @@ type State<'TModel,'TMsg> (initialState : 'TModel, update : 'TMsg -> 'TModel ->
         member this.Dispose() = 
             (signal :?> System.IDisposable).Dispose()            
             GC.SuppressFinalize this
+
+/// A thread-safe wrapper for a reference value with change notification
+type AtomicMutable<'a when 'a : not struct>(value : 'a) as self =
```

Comment:
Linter was giving me errors without the constraint, which might reduce restrictions with this next iteration. I'll try it without the constraint.